src/components/button/button.tsx

```diff
@@ -53,68 +55,59 @@ export class Button {
      * Please note that this does _not_ disable the button.
      * If the button should be disabled while loading, the
      * `disabled` property should be set to `true` as well.
+     *
+     * @deprecated Use limel-loading-button instead.
      */
     @Prop({ reflect: true })
     public loading = false;
 
     @Element()
-    private limelButton: HTMLLimelButtonElement;
+    private host: HTMLLimelButtonElement;
 
-    private justLoadedTimeout: NodeJS.Timeout;
+    public componentDidLoad() {
+        this.loadingWatcher(this.loading, this.loading);
+    }
+
+    public connectedCallback() {
+        this.loadingWatcher(this.loading, this.loading);
+    }
 
     public render() {
         return (
-            <button
-                class={`
-                    mdc-button
-                    ${this.primary ? 'mdc-button--unelevated' : ''}
-                    ${this.loading ? 'loading' : ''}
-                    ${this.outlined ? 'mdc-button--outlined' : ''}
-                `}
+            <limel-loading-button
```

Comment:
> Having `limel-button` wrap `limel-loading-button` seems very odd to me. The loading button is a button with some added functionality, not the other way around.

Yeah, I was thinking along the same lines to begin with, but then I hit the problem of how to do the spinner and stuff…

> I guess that could be done if limel-button rendered a <slot> for the spinner and svg.

I hadn't thought of that possibility, but regardless of whether we use a slot, a prop, or a method, we need to add _something_ to the interface of `limel-button` that lets `limel-loading-button` control stuff inside `limel-button`. And that's exactly what I would like to avoid… Maybe there's some other way? I guess we maybe could use an event of some sort…? 🤔

Reply:
Hmm… no, maybe a slot would be a pretty nice way to do it… 🤔

Reply:
And yes, _undocumented_ in that case 😄